### PR TITLE
Apply PHP 7.4 syntax and typed property

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 7.4, php 7.4 syntax can be applied with typed properties. 

- For `final class`, typed properties for protected is allowed if no inherit of parent or used by its final class parent.
- Otherwise, update only private properties.

### CODE REVIEW

**Comment by samsonasik** [comment]
All green 🎉

File: src/FileInput.php [hunk 1]
```diff
@@ -136,7 +136,9 @@ public function isValid($context = null)
             return true;
         }
 
-        return $this->implementation->isValid($context);
+        /** @var FileInputDecoratorInterface $implementation */
+        $implementation = $this->implementation;
+        return $implementation->isValid($context);
```
Comment (gsteel) [inline src/FileInput.php]: Nitpick, but this could be `assert($this->implementation !== null);`

File: src/FileInput.php [hunk 2]
```diff
@@ -136,7 +136,9 @@ public function isValid($context = null)
             return true;
         }
 
-        return $this->implementation->isValid($context);
+        /** @var FileInputDecoratorInterface $implementation */
```
Comment (Ocramius) [inline src/FileInput.php]: This should be an assertion: `assert(... !== null)`

**Review by Ocramius — APPROVED** [review]

File: src/FileInput.php (hunk 1 above)
Comment (samsonasik) [reply src/FileInput.php]: updated https://github.com/laminas/laminas-inputfilter/pull/67/commits/987edc5398e16bda34ea0c5ea4b1f288a2f90e63

File: src/FileInput.php (hunk 2 above)
Comment (samsonasik) [reply src/FileInput.php]: updated https://github.com/laminas/laminas-inputfilter/pull/67/commits/987edc5398e16bda34ea0c5ea4b1f288a2f90e63

**Comment by Ocramius** [comment]
Thanks @samsonasik!

**Pull request merged**